### PR TITLE
[SDK-2584] Implement new field filtering and pagination helper

### PR DESCRIPTION
**Pull request by evansims**
This PR includes a number of misc tweaks and improvements in preparation for SDK 8.0:

- Encompass the work in SDK-2584, finishing implementing and finalizing the API for the pagination and field filtering helpers on RequestOptions, used in the Management API.
- Moves several remaining /Helpers/* classes into the new /Utility/* namespace.
- Removes the obsolete header class types and tests around those; this is now handled with simple string arrays.
- Uses multi-byte safe string comparators.
- Removes the setContentType parameter from HttpClient::method(), as this is no longer needed.
- Ensures classes are defined 'final' (non-extendable) and use private var types as appropriate.
- Ensure exceptions all extend the SdkException type for easy error trapping in host applications.
- Renames HttpResponse::getJson() to ::decodeContent() for clarity (as we're really decoding/parsing a JSON response, not returning JSON.)
- Renames GenericResource class that Management endpoint classes extend to ManagementResource for clarity.
- Clean up Management token handling.

### CODE REVIEW

**Comment by evansims** [comment]
@adamjmcgrath I've introduced token expiration tracking per your previous suggestion, it's now stored in the session and accessible from Auth0::getTokenExpiration(). Let me know if you think this approach is satisfactory

**Comment by evansims** [comment]
Here's an example of how one could use all aspects of the new interface in PHP 8:

```php
$fields = new FilteredRequest(
    fields: ['created_at'],
    includeFields: true
);

$pagination = new PaginatedRequest(
    page: 0,
    perPage: 5,
    includeTotals: true
);

$options = new RequestOptions(
    fields: $fields,
    pagination: $pagination
);

$request = $managementApi->users()->getAll(
    options: $options
);
```

Or, more concisely:

```php
$request = $managementApi->users()->getAll(
    options: new RequestOptions(new FilteredQuest(['created_at'], true), new PaginatedRequest(0, 5, true))
);
````

As this RequestOptions interface could be expanded to provide configurations for more situations, for future purposes it's available on all API endpoints, even if the endpoint doesn't support pagination or field filtering.

File: src/Auth0.php [hunk 1]
```diff
@@ -412,6 +437,23 @@ public function setRefreshToken(
...
+     *
+     * @param string $refreshToken - refresh token returned from the code exchange.
+     */
+    public function setTokenExpiration(
+        string $tokenExpiration
+    ): self {
+        $this->state->setTokenExpiration($tokenExpiration);
+
+        if ($this->configuration->hasSessionStorage() && $this->configuration->getPersistAccessToken()) {
+            $this->configuration->getSessionStorage()->set('tokenExpiration', $tokenExpiration);
```
Comment (adamjmcgrath) [inline src/Auth0.php:451]: Can we call it `accessTokenExpiration`? Expiration of the other tokens can be a source of confusion

**Review by adamjmcgrath — CHANGES_REQUESTED** [review]

File: src/Auth0.php (hunk 1 above)
Comment (evansims) [reply src/Auth0.php]: Yup, good callout there. I've updated the naming for clarity 👍

**Review by adamjmcgrath — APPROVED** [review]

**Pull request merged**